src/Datadog.Trace/Tracer.cs

```diff
@@ -117,6 +118,13 @@ internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler samp
                 _agentWriter = new AgentWriter(api, Statsd, queueSize: Settings.TraceQueueSize);
             }
 
+            switch (Settings.Convention)
+            {
+                default:
+                    OutboundHttpConvention = new DatadogOutboundHttpConvention(this);
```

Comment:
the bi-directional reference is ugly...

Reply:
hmmm now I see the reason of the .ctor with the tracer object.

It seems weird to me storing data from integrations (`IOutboundHttpConvention`
 in this case) inside the tracer, this means that if we add new integrations that support conventions we need to modify the tracer as well?

Consider to expose only the `ConventionType` from the settings and each integration (that supports conventions) reads this value from the current `Tracer.Instance` and configure itself. 

I prefer decouple the integrations from the tracer.

Reply:
Relying on `Tracer.Instace` would make the code less maintainable/readable/testable.

I think that this proposal: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/63/files#r585368841 together with https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/63/files#r585374238 will address your main concerns.

The `Tracer` is the root object (aggregate) so it is normal that something "inside" would need to be changed when adding new stuff. Creating a facade would at least do it not "directly".

Maybe extracting some of the Tracer's functionalities to create spans would make it a little cleaner, but TBH I am not sure if it is worth the effort.

Reply:
Good thanks for clarifying.